greenheart/tools/eco/hydrogen_mgmt.py

```diff
@@ -294,6 +298,7 @@ def run_h2_storage(
             hydrogen_storage_capacity_kg, hydrogen_storage_duration_hr, hydrogen_storage_soc = hydrogen_storage_capacity(electrolyzer_physics_results['H2_Results'], greenheart_config['electrolyzer']['rating'], hydrogen_storage_demand)
             greenheart_config["h2_capacity"] = hydrogen_storage_capacity_kg
```

Comment:
We need to not save results to the config dict because the config dict is persistent across runs that do more than one analysis (optimization, design of experiments, step size studies). Results should only go to results dicts or classes